src/app/core/session/session.module.ts

```diff
@@ -39,27 +31,17 @@ import { serviceProvider } from "../../utils/utils";
  * [Session Handling, Authentication & Synchronisation]{@link /additional-documentation/concepts/session-and-authentication-system.html}
  */
 @NgModule({
+  imports: [KeycloakAngularModule],
   providers: [
-    SyncedSessionService,
+    SessionManagerService,
     LocalSession,
-    RemoteSession,
-    serviceProvider(SessionService, (injector: Injector) => {
-      return environment.session_type === SessionType.synced
-        ? injector.get(SyncedSessionService)
-        : injector.get(LocalSession);
-    }),
     KeycloakAuthService,
-    CouchdbAuthService,
-    serviceProvider(AuthService, (injector: Injector) => {
-      return environment.authenticator === AuthProvider.Keycloak
-        ? injector.get(KeycloakAuthService)
-        : injector.get(CouchdbAuthService);
-    }),
-    {
-      provide: HTTP_INTERCEPTORS,
-      useClass: AuthInterceptor,
-      multi: true,
-    },
+    LoginStateSubject,
+    SyncStateSubject,
```

Comment:
I really like the simplicity in tests and other places to access these states now! However, I do wonder if it wouldn't be easier to find the relevant state (for someone less familiar with the code base) if we do collect them in a single provided service (containing loginState + syncState + currentUser)?